### PR TITLE
enhance: update cagra index params in config and add params check

### PR DESCRIPTION
**Pull request by cqy123456**
issue:https://github.com/milvus-io/milvus/issues/29230
this pr do two things about cagra index:
 a.milvus yaml config support gpu memory settings

 b.add cagra-params check

### CODE REVIEW

**Review by yah01 — COMMENTED** [review]
/lgtm

**Review by presburger — APPROVED** [review]

**Comment by cqy123456** [comment]
rerun ut

File: internal/core/src/config/ConfigKnowhere.cpp [hunk 1]
```diff
@@ -88,6 +88,16 @@ KnowhereInitSearchThreadPool(const uint32_t num_threads) {
     }
 }
 
+void
+KnowhereInitGPUMemoryPool(const uint32_t init_size, const uint32_t max_size) {
+    if (init_size == 0 && max_size == 0) {
+        knowhere::KnowhereConfig::SetRaftMemPool();
+        return;
+    }
+    knowhere::KnowhereConfig::SetRaftMemPool(size_t{init_size},
```
Comment (czs007) [inline internal/core/src/config/ConfigKnowhere.cpp:97]: What would happen if one of these variables is equal to zero?

Comment (Presburger) [reply internal/core/src/config/ConfigKnowhere.cpp:97]: If set to 0, the GPU memory pool will directly occupy half of the total GPU memory.

**Pull request merged**